e2e/logs_catcher/matcher.go

```diff
@@ -2,118 +2,269 @@ package logs_catcher
 
 import (
 	"context"
-	"fmt"
-	"time"
-
+	"encoding/json"
 	"errors"
+	"fmt"
 	"go.uber.org/zap"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
 
 	"github.com/bloxapp/ssv/e2e/logs_catcher/docker"
 	"github.com/bloxapp/ssv/e2e/logs_catcher/logs"
 )
 
-// Test conditions:
+const beaconProxyContainer = "beacon_proxy"
+
+var ssvNodesContainers = []string{"ssv-node-1", "ssv-node-2", "ssv-node-3", "ssv-node-4"}
+
+type Matcher struct {
+	logger *zap.Logger
+	cli    DockerCLI
+	mode   SubMode
+}
+
+func NewLogMatcher(logger *zap.Logger, cli DockerCLI, mode SubMode) *Matcher {
+	return &Matcher{
+		mode:   mode,
+		cli:    cli,
+		logger: logger,
+	}
+}
+
+// Match is the entry point to start log catching based on the mode
+func (m *Matcher) Match(pctx context.Context) error {
+	ctx, cancel := context.WithTimeout(pctx, time.Minute*6*4) // wait max 4 epochs
+	defer cancel()
 
-const waitTarget = "beacon_proxy"
-const firstTarget = "beacon_proxy"
+	if _, err := m.waitForStartCondition(ctx, []string{waitFor}, beaconProxyContainer); err != nil {
+		return err
+	}
 
-var secondTargets = []string{"ssv-node-1", "ssv-node-2", "ssv-node-3", "ssv-node-4"}
+	return m.processMode(ctx)
+}
 
-const waitFor = "End epoch finished"
+func (m *Matcher) processMode(ctx context.Context) error {
```

Comment:
I don't get this func, all it does is to check that the passed mode exits? IMO should be done on a much higher level than in this impl

Reply:
it would make more sense make to extract each mode to a different function and return it based on the mode here.